src/components/Main/Main.js

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from "react";
+import TaskList from "../TaskList/TaskList";
+import CreateTask from "../CreateTask/CreateTask";
+import UpdateTask from "../UpdateTask/UpdateTask";
+import DeleteTask from "../DeleteTask/DeleteTask";
+//import Task from "../Task/Task";
+//import { useNavigate } from "react-router-dom";
+import { deleteTask, createTask, updateTask } from "../../utils/api";
+
+const Main = () => {
+  const [tasks, setTasks] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedTask, setSelectedTask] = useState(null);
+  const [activeModal, setActiveModal] = useState("");
+
+  const handleCloseModal = () => {
+    setActiveModal("");
+  };
+
+  const handleSelectedTask = (task) => {
+    setActiveModal("preview");
+    setSelectedTask(task);
+  };
+
+  const handleDeleteTask = (selectedTask) => {
+    setIsLoading(true);
+    deleteTask(selectedTask.id)
+      .then(() => {
+        const updatedTasks = tasks.filter(
+          (task) => task.id !== selectedTask.id
+        );
+        setTasks(updatedTasks);
+        handleCloseModal();
+      })
+      .catch((err) => {
+        console.error("Error deleting task:", err);
+      })
+      .finally(() => setIsLoading(false));
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js
  function handleSubmit(request) {
    // start loading
    setIsLoading(true);
    request()
      // we need to close only in `then`
      .then(handleCloseModal) 
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
      .catch(console.error)
      // and in finally we need to stop loading
      .finally(() => setIsLoading(false));
  }

```


You need to pass the **request** into the call of `handleSubmit` like that:

```js

  // here is an example
  const handleAddItem = (item) => {
    // here we create a function that returns a promise
    const makeRequest = () => {
      // `return` lets us use a promise chain `then, catch, finally`
      return addNewItem(item).then((item) => {
        setClothingItems([item, ...clothingItems]);
      });
    };
    // here we call handleSubmit passing the request
    handleSubmit(makeRequest);
  };

```

And for the 14th sprint:

```js
 // here is an example
 function handleProfileFormSubmit(inputValues) {
  // here we create a function that returns a promise 
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally`
    return api.editProfile(inputValues).then(setCurrentUser);
  }
  // here we call handleSubmit passing the request
  handleSubmit(makeRequest);
}


```

So, this way you can remove a lot of code duplicating